### PR TITLE
847-react-typescript-examples

### PR DESCRIPTION
**Pull request by clayinfor**
**Explain the details for making this change. What existing problem does the pull request solve?**

Add React Typescript examples.

**Related github/jira issue(s) (required)**:

Closes https://github.com/infor-design/enterprise-wc/issues/847

**Steps necessary to review your pull request (required)**:

1. Pull down repo and check out branch `847-react-typescript-examples`
2. Do: `cd react-ts-ids-wc`
3. Do: `npm install`
4. Do: `npm run start`
5. See http://localhost:3000/

### CODE REVIEW

**Review by andriivarhanov — COMMENTED** [review]
looks good, I think we can maybe cleanup CRA generated files and replace them with ours (readme, logos, manifest, App.css, test etc)

**Review by tmcconechy — DISMISSED** [review]
Looks great! I like @andriivarhanov suggestion if you can

**Comment by clayinfor** [comment]
Hey @andriivarhanov and @tmcconechy, thanks for the feedback!  This is now good to go.

**Review by andriivarhanov — APPROVED** [review]
looks good 👍

File: react-ts-ids-wc/src/react-app-env.d.ts
Comment (clayinfor) [inline react-ts-ids-wc/src/react-app-env.d.ts:1]: hey @tmcconechy , one thing to note is if we add the new `src/declarations.d.ts` file I added here to our `enterprise-wc` repository, then React users can simply add the following line to this `react-app-env.dt.s` to get things to just work... and they won't have to keep track of which components we add or remove.

`/// <reference types="ids-enterprise-wc" />`

**Pull request merged**